lib/queue.js

```diff
@@ -276,7 +276,7 @@ Queue.prototype.push = function (input, cb) {
         if (err) return ticket.failed('id_error');
         acceptTask(id);
       })
-    } else if (typeof self.id === 'string' && typeof task === 'object') {
+    } else if (typeof self.id === 'string' && typeof task === 'object' && task[self.id] !== undefined ) {
```

Comment:
I don't think this is required -- if `task[self.id]` is undefined it's the same as calling `acceptTask(undefined)`